### PR TITLE
Implemented Cyphal/CAN as a "Decode As" Filter

### LINKED ISSUE #2: Import Cyphal/CAN from older UAVCAN filter
The older filter was written in C so a new filter will have to be rewritten from scratch in Lua if we want that. UDP was very easy in Lua so a working CAN version can't be too hard.

### PR DESCRIPTION
**Pull request by emrainey**
Adding Plugin info.
Fixes #2

### CODE REVIEW

File: cyphal_can.lua [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    -- Tail Byte Decode
+    local tail_byte = tail_buffer:uint()
+    footer_tree:add(cyphal_can_start_of_transfer, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_end_of_transfer, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_toggle, tail_buffer, tail_byte)
+    footer_tree:add(cyphal_can_transfer_id, tail_buffer, tail_byte)
+    local sot = bit.rshift(bit.band(tail_byte, 0x80), 7)
+    local eot = bit.rshift(bit.band(tail_byte, 0x40), 6)
+    if sot == 0 and eot == 1 then
+      payload_tree:add(cyphal_can_crc, payload(payload_len - 2, 2))
```
Comment (pavel-kirienko) [inline cyphal_can.lua:164]: Is this correct, considering that the CRC is encoded in the big-endian format?

File: cyphal_can.lua [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+}
+cyphal_heartbeat_health = ProtoField.uint8("cyphal.heartbeat.health", "heartbeat.health", base.DEC, healthTable)
+local modeTable = {
+    [0] = "Operational",
+    [1] = "Initialization",
+    [2] = "Maintenance",
+    [3] = "SoftwareUpdate"
+}
+cyphal_heartbeat_mode = ProtoField.uint8("cyphal.heartbeat.mode", "heartbeat.mode", base.DEC, modeTable)
+cyphal_heartbeat_vssc = ProtoField.uint16("cyphal.heartbeat.vssc", "heartbeat.vssc", base.HEX)
```
Comment (pavel-kirienko) [inline cyphal_can.lua]:
```suggestion
cyphal_heartbeat_vssc = ProtoField.uint8("cyphal.heartbeat.vssc", "heartbeat.vssc", base.HEX)
```

File: cyphal_can.lua [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+cyphal_getinfo_hardware_version_major = ProtoField.uint8("cyphal.getinfo.response.hardware_version.major", "hardware_version.major", base.DEC)
+cyphal_getinfo_hardware_version_minor = ProtoField.uint8("cyphal.getinfo.response.hardware_version.minor", "hardware_version.minor", base.DEC)
+cyphal_getinfo_software_version_major = ProtoField.uint8("cyphal.getinfo.response.software_version.major", "software_version.major", base.DEC)
+cyphal_getinfo_software_version_minor = ProtoField.uint8("cyphal.getinfo.response.software_version.minor", "software_version.minor", base.DEC)
+cyphal_getinfo_software_vcs_revision_id = ProtoField.uint64("cyphal.getinfo.response.cyphal_getinfo_software_vcs_revision_id", "SW VCS Revision ID", base.HEX)
+cyphal_getinfo_unique_id = ProtoField.bytes("cyphal.getinfo.response.unique_id", "Node Unique ID")
+cyphal_getinfo_name = ProtoField.string("cyphal.getinfo.response.name", "Node Name")
+cyphal_getinfo_software_image_crc = ProtoField.uint64("cyphal.getinfo.response.crc", "CRC-64-WE", base.HEX)
+cyphal_getinfo_certificate_of_authority = ProtoField.bytes("cyphal.getinfo.response.certificate_of_authority", "Certificate of Authority")
```
Comment (pavel-kirienko) [inline cyphal_can.lua]:
```suggestion
cyphal_getinfo_certificate_of_authenticity = ProtoField.bytes("cyphal.getinfo.response.certificate_of_authenticity", "Certificate of Authenticity")
```

File: cyphal_can.lua (hunk 1 above)
Comment (emrainey) [reply cyphal_can.lua:164]: Lua Wireshark filters are by default BE unless you specifically add as LE.

File: cyphal_can.lua [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    [3] = "High",
+    [4] = "Nominal",
+    [5] = "Low",
+    [6] = "Slow",
+    [7] = "Optional"
+}
+cyphal_can_priority = ProtoField.uint32("cyphal_can.priority", "Priority", base.DEC, priorityTable, 0x1C000000, "Priority of the Transfer")
+cyphal_can_len = ProtoField.uint8("cyphal_can.payload_length", "Payload Length", base.DEC)
+local service_or_messageTable = {
+    [0] = "Broadcast Message",
```
Comment (thirtytwobits) [inline cyphal_can.lua]: "Message Transfer"

File: cyphal_can.lua [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    [1] = "Service Oriented Message"
```
Comment (thirtytwobits) [inline cyphal_can.lua]: "Service Transfer"

File: cyphal_can.lua [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+cyphal_can_subject_id = ProtoField.uint32("cyphal_can.subject_id", "subject_id", base.DEC, nil, 0x001FFF00)
+cyphal_can_service_id = ProtoField.uint32("cyphal_can.service_id", "service_id", base.DEC, nil, 0x007FC000)
+cyphal_can_destination_node_id = ProtoField.uint32("cyphal_can.destination_node_id", "destination_node_id", base.DEC, nil, 0x0003F80)
+cyphal_can_source_node_id = ProtoField.uint32("cyphal_can.source_node_id", "source_node_id", base.DEC, nil, 0x0000007F)
+cyphal_can_start_of_transfer = ProtoField.uint8("cyphal_can.start_of_transfer", "start_of_transfer", base.DEC, nil, 0x80)
+cyphal_can_end_of_transfer = ProtoField.uint8("cyphal_can.end_of_transfer", "end_of_transfer", base.DEC, nil, 0x40)
+cyphal_can_toggle = ProtoField.uint8("cyphal_can.toggle", "toggle", base.DEC, nil, 0x20)
+cyphal_can_transfer_id = ProtoField.uint8("cyphal_can.transfer_id", "transfer_id", base.DEC, nil, 0x1F)
+cyphal_can_tail_byte = ProtoField.uint8("cyphal_can.tail_byte", "tail_byte", base.HEX)
+cyphal_can_crc = ProtoField.uint16("cyphal_can.crc16", "crc16", base.HEX)
```
Comment (thirtytwobits) [inline cyphal_can.lua]: CRC-16/CCITT-FALSE

**Review by thirtytwobits — APPROVED** [review]
Just some naming nits.

File: cyphal_can.lua (hunk 6 above)
Comment (emrainey) [reply cyphal_can.lua]:
```suggestion
cyphal_can_crc = ProtoField.uint16("cyphal_can.crc16", "CRC16/CCITT-FALSE", base.HEX)
```

File: cyphal_can.lua (hunk 4 above)
Comment (emrainey) [reply cyphal_can.lua]:
```suggestion
    [0] = "Message Transfer",
```

File: cyphal_can.lua (hunk 5 above)
Comment (emrainey) [reply cyphal_can.lua]:
```suggestion
    [1] = "Service Transfer"
```

**Pull request merged**